lib/CMakeLists.txt

```diff
@@ -1,6 +1,8 @@
 add_subdirectory(core)
 add_subdirectory(render)
 
-add_library(Airship)
+add_library(Airship INTERFACE)
 
-target_link_libraries(Airship PUBLIC core AirshipRenderer)
+target_link_libraries(Airship INTERFACE AirshipCore AirshipRenderer)
+add_custom_target(AllAirshipLibs)
```

Comment:
Why have `Airship` and `AllAirshipLibs` separately? At least, `Airship` should link against `AllAirshipLibs` instead of linking directly to each library - unless we can merge these two targets which would be more ideal to me.